--- a/conf/conf_user_def.inc.php
+++ b/conf/conf_user_def.inc.php
@@ -40,4 +40,4 @@
 // ("http://menu.2ch.net/bbsmenu.html")                     // 2chŠî–{

-$conf_user_def['brdfile_online'] = "http://azlucky.s25.xrea.com/2chboard/bbsmenu.html";
+$conf_user_def['brdfile_online'] = "http://https://menu.5ch.net/bbsmenu.html";
 $conf_user_rules['brdfile_online'] = array('emptyToDef', 'invalidUrlToDef');
