--- a/conf/conf_user_def.inc.php
+++ b/conf/conf_user_def.inc.php
@@ -39,7 +39,7 @@
 // ("http://azlucky.s25.xrea.com/2chboard/bbsmenu.html")    // 2ch + ŠO•”BBS
 // ("http://menu.2ch.net/bbsmenu.html")                     // 2chŠî–{

-$conf_user_def['brdfile_online'] = "http://azlucky.s25.xrea.com/2chboard/bbsmenu.html";
+$conf_user_def['brdfile_online'] = "https://menu.5ch.net/bbsmenu.html";
 $conf_user_rules['brdfile_online'] = array('emptyToDef', 'invalidUrlToDef');

 // }}}
