--- a/p2-php/conf/conf.inc.php
+++ b/p2-php/conf/conf.inc.php
@@ -144,3 +144,3 @@

-    define('P2_CONFIG_DIR', __DIR__);
+    define('P2_CONFIG_DIR', '/var/www/conf');

--- a/p2-php/conf/conf_user_def.inc.php
+++ b/p2-php/conf/conf_user_def.inc.php
@@ -529,10 +529,10 @@
 // プロキシを利用(する:1, しない:0)
-$conf_user_def['proxy_use'] = 0; // (0)
+$conf_user_def['proxy_use'] = 1; // (0)
 $conf_user_rad['proxy_use'] = array('1' => 'する', '0' => 'しない');

 // プロキシホスト ex)"127.0.0.1", "www.p2proxy.com"
-$conf_user_def['proxy_host'] = ""; // ("")
+$conf_user_def['proxy_host'] = "127.0.0.1"; // ("")

 // プロキシポート ex)"8080"
-$conf_user_def['proxy_port'] = ""; // ("")
+$conf_user_def['proxy_port'] = "8080"; // ("")

--- a/p2-php/conf/conf_admin_ex.inc.php
+++ b/p2-php/conf/conf_admin_ex.inc.php
@@ -103,3 +103,3 @@
 // ImageCache2（off:0, PCのみ:1, 携帯のみ:2, 両方:3）
-$_conf['expack.ic2.enabled'] = 0; // (0)
+$_conf['expack.ic2.enabled'] = 3; // (0)

--- a/p2-php/conf/conf_ic2.inc.php
+++ b/p2-php/conf/conf_ic2.inc.php
@@ -19,3 +19,3 @@
 // 注2: MySQL,PosrgreSQLでは予めデータベースを作っておく。
-$_conf['expack.ic2.general.dsn'] = "";
+$_conf['expack.ic2.general.dsn'] = 'sqlite:' . $_conf['db_dir'] . '/image_cache.sqlite';

@@ -40,3 +40,3 @@
 // 明示的に"ImageMagick6"を指定しなくてもよい
-$_conf['expack.ic2.general.driver'] = "gd";
+$_conf['expack.ic2.general.driver'] = "imagick";

